client/app/components/BaseContainer/index.jsx

```diff
@@ -1,99 +1,76 @@
 // @flow
-import React from 'react';
+import React, { Fragment } from 'react';
 import axios from 'axios';
 import { StoryContainer } from './StoryContainer';
+import css from '../../styles/_global.scss';
+import { I18n } from '../../libs/i18n';
 
 export type Props = {
   container: string,
   data: any,
   fetchUrl: string,
-  isLastPage?: boolean,
+  lastPage?: boolean,
 };
 
 export type State = {
-  loading: boolean,
-  isLastPage: boolean,
+  lastPage: boolean,
   page: number,
   data: any,
 };
 
 export class BaseContainer extends React.Component<Props, State> {
   constructor(props: Props) {
     super(props);
+    const { data, lastPage } = props;
     this.state = {
       page: 1,
-      isLastPage: props.isLastPage || false,
-      loading: false,
-      data: props.data,
+      lastPage: !!lastPage,
+      data,
     };
   }
 
-  componentDidMount() {
-    window.addEventListener('scroll', this.handleScroll);
-  }
-
-  componentDidUpdate() {
-    const { isLastPage } = this.state;
-    if (isLastPage) {
-      window.removeEventListener('scroll', this.handleScroll);
-    }
-  }
-
-  componentWillUnmount() {
-    window.removeEventListener('scroll', this.handleScroll);
-  }
-
-  calculateDocument = () => {
-    if (document.documentElement) {
-      return (
-        document.documentElement.offsetHeight
-        - document.documentElement.scrollTop
-      );
-    }
-    return 0;
-  };
-
-  handleScroll = () => {
+  onClick = () => {
+    const { page, data } = this.state;
     const { fetchUrl } = this.props;
-    const { page, loading, data } = this.state;
-    if (
-      !loading
-      && document.getElementsByClassName('footer')[0].clientHeight
-        + window.innerHeight
-        > this.calculateDocument()
-    ) {
-      const url = `${window.location.origin + fetchUrl}.json?page=${page + 1}`;
-      this.setState({ loading: true });
-      axios
-        .get(url)
-        .then((response: any) => {
-          this.setState({
-            loading: false,
-            page: page + 1,
-            data: data.concat(response.data.data),
-            isLastPage: response.data.lastPage,
-          });
-        })
-        .catch((error) => {
```

Comment:
Removing the catch block because we want to throw the error from the server. The value of loading will already be false if an error is thrown.